#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/CircleArena.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.11569254522180486 -0.5781288097347712 0.8077018721872306 0.8314361947163936
  position 0.07507026871885247 0.6107323214593312 -0.029117679521809135
}
TexturedBackground {
}
TexturedBackgroundLight {
}
CircleArena {
}
Robot {
  translation 0.33342989466220246 0.8678345116104149 0.06610659970005149
  rotation 0.9650365724876662 -0.2612898376712772 0.02078543939816571 -2.366556611958606
  children [
    Gyro {
      translation 0 -0.01 -0.04
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 1 0 0.498039
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
    }
    Solid {
      translation 0 0 0.024
      children [
        DEF holder Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.045 0.15 0.01
          }
        }
      ]
      name "holder"
      boundingObject USE holder
      physics Physics {
        density -1
        mass 1
      }
    }
    Solid {
      translation 0 0.025 -0.022
      rotation 1 0 0 1.57
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Cylinder {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "head"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 9018.833781637379
        anchor 0 0 0.04
      }
      device [
        RotationalMotor {
          name "motor1"
        }
      ]
      endPoint DEF wheel1 Solid {
        translation -0.0399728273342404 0.00227453267280313 0.037398819813797134
        rotation -0.10519127672077273 -0.9888950486936673 -0.10498275082604432 4.702102051530119
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
                url [
                  "../imgs/1f974.png"
                ]
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    InertialUnit {
      translation 0 0 -0.04
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
          }
          geometry Box {
            size 0.005 0.005 0.005
          }
        }
      ]
      name "imu"
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 6857.019275935228
        anchor 0 0 0.04
      }
      device [
        RotationalMotor {
          name "motor2"
        }
      ]
      endPoint DEF wheel1 Solid {
        translation 0.039982831982230196 -0.0027098777153883374 0.04219536724112737
        rotation -0.7034583914641952 0.10560181746667512 -0.702847456868375 2.9303292964391825
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColorMap ImageTexture {
                url [
                  "../imgs/1f974.png"
                ]
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.03
            }
          }
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
          density -1
          mass 1
        }
      }
    }
    DEF body Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
      }
      geometry Box {
        size 0.05 0.05 0.07
      }
    }
  ]
  boundingObject USE body
  physics Physics {
    density -1
    mass 10
  }
  controller "ccc"
  controllerArgs [
    ""
  ]
  linearVelocity 0.0018294018285202598 0.004972010610607122 -0.0023960292353193908
  angularVelocity -0.07005941202978122 0.028104578200429935 0.004228434672358639
}
